addons/voice/org.openhab.voice.voicerss/src/main/java/org/openhab/voice/voicerss/internal/cloudapi/VoiceRSSCloudImplementation.java

```diff
@@ -0,0 +1,149 @@
+ * <li>Only default voice supported with good audio quality</li>
+ * <li>Only MP3 audio format supported</li>
+ * <li>It uses HTTP and not HTTPS (for performance reasons)</li>
+ * </ul>
+ * 
+ * @author Jochen Hiller - Initial contribution
+ */
+public class VoiceRSSCloudImplementation implements VoiceRSSCloudAPI {
+
+	private final Logger logger = LoggerFactory.getLogger(VoiceRSSCloudImplementation.class);
+
+	/**
+	 * Will support only "MP3" for the moment.
+	 */
+	@Override
+	public Set<String> getAvailableAudioFormats() {
+		Set<String> formats = new HashSet<String>();
+		formats.add("MP3");
+		return formats;
+	}
+
+	/**
+	 * Will support only 3 locales for the moment.
+	 */
+	@Override
+	public Set<Locale> getAvailableLocales() {
+		Set<Locale> locales = new HashSet<Locale>();
+		locales.add(Locale.forLanguageTag("en-us"));
+		locales.add(Locale.forLanguageTag("en-gb"));
+		locales.add(Locale.forLanguageTag("de-de"));
+		return locales;
+	}
+
+	/**
+	 * Will support only a default voice with good quality.
+	 */
+	@Override
+	public Set<String> getAvailableVoices() {
+		// only one default voice, for every locale
+		Set<String> voices = new HashSet<String>();
+		voices.add("default");
+		return voices;
+	}
+
+	/**
+	 * The default voice will be supported for all locales.
+	 */
+	@Override
+	public Set<String> getAvailableVoices(Locale locale) {
+		return getAvailableVoices();
+	}
+
+	/**
+	 * This method will return an input stream to an audio stream for the given
+	 * parameters.
+	 * 
+	 * It will do that using a plain URL connection to avoid any external
+	 * dependencies.
+	 */
+	@Override
+	public InputStream getTextToSpeech(String apiKey, String text, String locale, String audioFormat)
+			throws IOException {
+		String url = createURL(apiKey, text, locale, audioFormat);
+		logger.info("Call {}", url);
+		URLConnection connection = new URL(url).openConnection();
+		// mark our call to be visible where we make the requests
+		connection.setRequestProperty("user-agent", "EclipseSmartHome/1.0");
+
+		// we will check return codes. The service will ALWAYS return a HTTP
+		// 200, but for error messages, it will return a text/plain format and
+		// the error message in body
+		int status = ((HttpURLConnection) connection).getResponseCode();
+		if (HttpURLConnection.HTTP_OK != status) {
+			logger.error("Call {} returned HTTP {}", status);
+			throw new IOException("Could not read from service: HTTP code" + status);
+		}
+		if (logger.isTraceEnabled()) {
+			for (Entry<String, List<String>> header : connection.getHeaderFields().entrySet()) {
+				logger.trace("Response.header: {}={}", header.getKey(), header.getValue());
+			}
+		}
+		String contentType = connection.getHeaderField("Content-Type");
+		InputStream is = connection.getInputStream();
+		// check if content type is text/plain, then we have an error
+		if (contentType.contains("text/plain")) {
+			byte[] bytes = new byte[256];
+			is.read(bytes, 0, 256);
+			throw new IOException(
```

Comment:
when you return with an exception, the caller won't have a chance to call is.close() - so it should be yourself, who does it.